src/coreclr/jit/importercalls.cpp

```diff
@@ -10811,7 +10868,14 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
                 {
                     namespaceName += 8;
 
-                    if (strcmp(namespaceName, "CompilerServices") == 0)
+                    if (strcmp(className, "TypeCast") == 0)
+                    {
+                        if (strcmp(methodName, "WriteBarrier") == 0)
+                        {
+                            result = NI_System_Runtime_CompilerServices_RuntimeHelpers_WriteBarrier;
+                        }
+                    }
+                    else if (strcmp(namespaceName, "CompilerServices") == 0)
```

Comment:
```suggestion
                    if (strcmp(namespaceName, "CompilerServices") == 0)
```
And move the WriteBarrier into RuntimeHelpers for NAOT instead? We prefer things to be the same between NAOT and non-NAOT where possible.

Reply:
@jkotas ah, sorry, I hit a different issue - so when `Runtime.Base` is compiled, it already has RuntimeHelpers, I presume because it's compiled against corelib/facade and I can't add a new RuntimeHelpers.cs to that
```
C:\prj\runtime-main2\src\coreclr\nativeaot\Runtime.Base\src\System\Runtime\TypeCast.cs(806,28): error CS0117: 
'RuntimeHelpers' does not contain a definition for 'WriteBarrier'
```

Reply:
Ah, ignore me, I think I fixed it 🙂